py/src/braintrust/framework.py

```diff
@@ -811,7 +829,7 @@ async def run_evaluator_task(datum):
             project=evaluator.project_name, experiment=base_experiment.name, open=True, set_current=False
         ).as_dataset()
 
-    if inspect.isfunction(data_iterator):
+    if inspect.isfunction(data_iterator) or not is_inspectable(data_iterator):
```

Comment:
maybe check `not is_inspectable(...)` first, in case the `isfunction` check throws up?

Reply:
In my testing it just returns false because the object isn't a python function, which is correct behavior.  More often than not the first clause will be true, which is why I left it first.  Happy to flip it though.